### PR TITLE
Fix qstat -pt crash with array jobs

### PR DESCRIPTION
**Pull request by vstumpf**
#### Bug/feature Description
* `qstat -pt` would result in a segmentation fault if 1 subjob with jobstate X was present in the output.
* `qstat -pt` would print (null) if the job was in the Q state.

#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* `percent_cal` was returning a string literal, and the calling function was freeing it.

#### Solution Description
* Allocate a character array and return it instead.

#### Testing logs/output
* [after-fix.txt](https://github.com/PBSPro/pbspro/files/2794306/after-fix.txt)
* [before-fix.txt](https://github.com/PBSPro/pbspro/files/2794307/before-fix.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/cmds/qstat.c [hunk 1]
```diff
@@ -1126,10 +1126,12 @@ percent_cal(char *state, char *timeu, char *timer, char *wtimu, char *wtimr, cha
 		case 'Q':
 		case 'T':
 		case 'W':
+			pbs_asprintf(&rtn, "%3s", "-- ");
```
Comment (mike0042) [inline src/cmds/qstat.c]: I expected to see a corresponding call to free(). I realize that qstat is supposed to run quickly and exit, but we should still be mindful of freeing dynamically allocated memory.

Comment (vstumpf) [reply src/cmds/qstat.c]: It does get freed by the caller of `percent_cal`: https://github.com/PBSPro/pbspro/blob/264cefe83eea3f51f5f84d77cd3a837e8dacbf46/src/cmds/qstat.c#L1430-L1432

Comment (mike0042) [reply src/cmds/qstat.c]: You are correct. Thanks for pointing it out.

**Comment by vstumpf** [comment]
@mike0042 I changed a bit of the regex, can you re-review?

**Comment by vstumpf** [comment]
Thanks, I'll rebase and squash

File: test/tests/functional/pbs_qstat.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
```
Comment (latha-subramanian) [inline test/tests/functional/pbs_qstat.py]: Should this be 2019?

**Comment by vstumpf** [comment]
Thanks for the review
As long as the maintainers are ok with it, I'd rather wait for the automated script to change it all.

**Comment by vstumpf** [comment]
Actually, since i have to rebase anwyays, I'll update the year.

**Comment by bhroam** [comment]
I merged in the PR.

**Pull request merged**